src/drawers/drawer.ts

```diff
@@ -43,23 +47,41 @@ export class Drawer {
     this._svgElementName = "path";
   }
 
+  public canvas(): d3.Selection<HTMLCanvasElement, any, any, any>;
+  /**
+   * Sets the renderArea() to null and tells this Drawer to draw to the specified canvas instead.
+   * @param canvas
+   */
+  public canvas(canvas: d3.Selection<HTMLCanvasElement, any, any, any>): this;
+  public canvas(canvas?: d3.Selection<HTMLCanvasElement, any, any, any>): any {
+    if (canvas === undefined) {
+      return this._canvas;
+    }
+    canvas = coerceExternalD3(canvas);
+    this._canvas = canvas;
+    this._renderArea = null;
+    this._cachedSelectionValid = false;
+    return this;
+  }
+
   /**
    * Retrieves the renderArea selection for the Drawer.
    */
   public renderArea(): SimpleSelection<void>;
   /**
-   * Sets the renderArea selection for the Drawer.
+   * Sets the canvas() to null and tells this Drawer to draw to the specified SVG area instead.
    *
    * @param {d3.Selection} Selection containing the <g> to render to.
    * @returns {Drawer} The calling Drawer.
    */
   public renderArea(area: SimpleSelection<void>): this;
   public renderArea(area?: SimpleSelection<void>): any {
-    if (area == null) {
+    if (area === undefined) {
       return this._renderArea;
     }
     area = coerceExternalD3(area);
     this._renderArea = area;
+    this._canvas = null;
```

Comment:
this is a strange side effect. Even with a comment, this could be troublesome if the canvas is set to null but not removed from the DOM, no?